File: 151051/Makovetskiy/task310+320/dc-rest/src/main/java/com/rmakovetskij/dc_rest/model/Issue.java
package com.rmakovetskij.dc_rest.model;

import jakarta.validation.constraints.Size;
import lombok.Data;

import jakarta.persistence.*; //поменял с javax на jakarta
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "tbl_issue")
public class Issue {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Size(min = 2, max = 64)
    @Column(unique = true, nullable = false)
    private String title;

    @Size(min = 4, max = 2048)
    @Column(nullable = false)
    private String content;


    @CreationTimestamp
    private LocalDateTime created;


    @UpdateTimestamp
    private LocalDateTime modified;

    // Ассоциативное поле, связь с Editor
    @ManyToOne
    @JoinColumn(name = "editor_id", nullable = false)
    private Editor editor;
}

File: 251003/Pankratiev/rest/src/main/kotlin/bsuir/dc/rest/entity/Issue.kt
package bsuir.dc.rest.entity

import java.time.LocalDateTime

data class Issue(
    override var id: Long = 0L,
    val writerId: Long,
    val title: String,
    val content: String,
    val created: LocalDateTime = LocalDateTime.now(),
    var modified: LocalDateTime = LocalDateTime.now(),
) : Identifiable
